src/view/Main/Main.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from 'react';
+import './Main.css';
+import Filter from '../../components/Filter/Filter';
+import Search from '../../components/Search/Search';
+import fetchCountries from '../../services/countries';
+import CountryCard from '../../components/CountryCard/CountryCard';
+
+export default function Main() {
+  const [countries, setCountries] = useState([]);
+  const [results, setResults] = useState([]);
+  const [selected, setSelected] = useState('All');
+  const [query, setQuery] = useState('');
+  const continents = [
+    'All',
+    'Africa',
+    'Asia',
+    'Europe',
+    'North America',
+    'South America',
+    'Oceania',
+  ];
+  useEffect(() => {
+    const fetch = async () => {
+      try {
+        const resp = await fetchCountries();
+        setCountries(resp);
+      } catch (error) {
+        alert(error.message);
+      }
+    };
+    fetch();
+  }, []);
+
+  useEffect(() => {
+    const filtered = countries.filter(
+      (country) => country.continent === selected || selected === 'All'
+    );
+    setResults(filtered);
+  }, [countries, selected]);
+
+  // filter function search should push a true if country.name(lowercase) includes query
+  const search = (arr) =>
+    arr.filter((country) => {
+      const name = country.name.toLowerCase();
+      if (name.includes(query)) {
+        return country;
+      } else if (query === '') {
+        return country;
+      }
+    });
```

Comment:
great job utilizing another useEffect in filtering the countries based on a the user query